pkg/reloader/reloader.go

```diff
@@ -94,41 +90,63 @@ func (r *Reloader) Run(ctx context.Context) error {
 	// Follow configuration updates in the directory where
 	// the config file is located and trigger reload when
 	// it is either recreated or written into.
-	if err := configWatcher.Add(path.Dir(r.ConfigFile)); err != nil {
-		return err
+	for i := range r.ConfigFiles {
+		// Ensure our paths are canonical
+		r.ConfigFiles[i], _ = filepath.Abs(r.ConfigFiles[i])
+		if err := configWatcher.Add(filepath.Dir(r.ConfigFiles[i])); err != nil {
```

Comment:
I was going to ask... Is this the directory because of the way k8s refreshes the files?  I assume it unmounts and remounts the entire dir?

Reply:
If that's the case, there should probably be a comment for posterity

Reply:
yes that’s right too... couldn’t make it work well in k8s unless using the directory